services/construction/construction_service.go

```diff
@@ -52,11 +54,24 @@ func NewAPIService(
 	errors []*types.Error,
 	client Client,
 ) *APIService {
+	// currently the eth client is only used for OP stack chains,
+	// we can remove the support check if it's used by multiple chains in the future
+	var ethClient *ethclient.Client
+	var err error
+	if cfg.RosettaCfg.SupportsOpStack {
+		ethClient, err = ethclient.Dial(cfg.GethURL)
+		if err != nil {
+			errString := fmt.Sprintf("failed to dial node %s", cfg.GethURL)
+			panic(errString)
+		}
+	}
+
 	return &APIService{
-		config: cfg,
-		types:  types,
-		errors: errors,
-		client: client,
+		config:    cfg,
+		types:     types,
+		errors:    errors,
+		client:    client,
+		ethClient: ethClient,
```

Comment:
I understand what you are doing, but this is not a very optimal solution, you are passing an ethClient only used it in a contract call.
Maybe we should implement an interface in `Client`, and have you op-stack chain implement this logic? Or any other alternatives?

Reply:
eth client is removed, we can leverage the existing eth client now!